src/main/java/org/craftercms/engine/util/spring/context/RestrictedApplicationContext.java

```diff
@@ -36,15 +37,24 @@
 
     public static final String CONFIG_KEY_BEAN_PATTERNS = "publicBeans.bean";
 
-    public RestrictedApplicationContext(ApplicationContext parent) {
+    protected List<String> defaultPublicBeans;
+
+    public RestrictedApplicationContext(ApplicationContext parent, List<String> defaultPublicBeans) {
         super(parent);
+        this.defaultPublicBeans = defaultPublicBeans;
     }
 
     protected boolean isAllowed(String name) {
-        HierarchicalConfiguration<?> siteConfig = ConfigUtils.getCurrentConfig();
-        List<String> beanPatterns = emptyList();
+        // allow access to everything during site initialization
```

Comment:
Why do we have to do this?

Reply:
When spring is creating the beans during site init there is no site context available and that affects beans being injected

Reply:
Can you add that as a comment, please? That way we won't forget why we added those lines :).